mlflow/pyfunc/__init__.py

```diff
@@ -1506,6 +1510,8 @@ def udf(
             os.environ.update(openai_env_vars)
         if mlflow_testing:
             _MLFLOW_TESTING.set(mlflow_testing)
+        if env_vars:
+            os.environ.update(env_vars)
```

Comment:
pyspark UDF worker might be reused by other spark jobs, see `spark.python.worker.reuse` config (default True), so we need to ensure the env is restored once UDF completes.